client/client.go

```diff
@@ -80,8 +80,8 @@ func (c *Client) RequestStreamWithHTTPClient(ctx context.Context, method string,
 
 	for _, inspector := range inspectors {
 		if err = inspector.InspectResponse(res); err != nil {
-			drainAndClose(res.Body)
-			return nil, err
+			log.LoggerFromContext(ctx).WithError(err).Warn("inspecting response")
```

Comment:
This change was the only red-flag for me in this PR, but I think it's ok, because `drainAndClose` is called in `handleResponse`. @darinkrauss is more familiar with the code than me, so I'll let him take a final a look.

Reply:
Yeah, that change isn't strictly required, but as they're written before this, response inspectors that return errors, will cause the entire request to return an error, regardless of whether the request was otherwise successful. Any additional inspectors (like possibly the metrics inspector being added in this PR) might not run at all, causing inaccurate metrics.

Reply:
If we log the error (rather than failing) then the `drainAndClose` isn't necessary as it should be done in `handleResponse`.

Though logging (versus failing) changes the nature of the protocol, I don't see a problem with changing it since there isn't any previous usage.

Reply:
Oh, and the `continue` is unnecessary.

Reply:
Hmm, I'm not sure I agree with this change. If an inspector is important and shouldn't  be affected by errors from other inspectors, then it should be listed first.

However, I won't cause my disagreement to block this PR.